js/ai/src/model-types.ts

```diff
@@ -406,5 +406,7 @@ export const GenerateActionOptionsSchema = z.object({
   returnToolRequests: z.boolean().optional(),
   /** Maximum number of tool call iterations that can be performed in a single generate call (default 5). */
   maxTurns: z.number().optional(),
+  /** Custom step name for this generate call to display in trace views. Defaults to "generate". */
+  stepName: z.string().optional(),
```

Comment:
These types need to be kept in sync with https://github.com/firebase/genkit/blob/main/genkit-tools/common/src/types/model.ts#L379